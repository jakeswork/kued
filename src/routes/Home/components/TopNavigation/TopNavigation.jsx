import React, { Component, Fragment } from 'react';
import { slide as Menu } from 'react-burger-menu';
import { FiMenu, FiX } from 'react-icons/fi';

import propTypes from './types';
import MenuItems from './components/MenuItems';
import Text from '../../../../components/Text';
import logo from '../../../../assets/kued-logotype.png';
import menuLogo from '../../../../assets/logo.png';
import Viewport from '../../../../utils/Viewport';

class TopNavigation extends Component {
  static propTypes = propTypes;

  state = {
    isMobileViewport: false,
    isOpen: false,
  }

  componentDidMount() {
    this.updateViewportState();
    window.addEventListener('resize', this.updateViewportState);
  }

  componentWillUnmount() {
    window.removeEventListener('resize', this.updateViewportState);
  }

  updateViewportState = () => {
    this.setState({ isMobileViewport: Viewport.mobile() });
  }

  render() {
    const { classes } = this.props;
    const { isMobileViewport, isOpen } = this.state;

    return (
      <Fragment>
        <img src={logo} alt="Kued Logotype" className={classes.img} />
        {
          isMobileViewport
            ? (
              <Fragment>
                <FiMenu
                  className={classes.menuIcon}
                  data-test-id="openMenuButton"
                  onClick={() => this.setState({ isOpen: !isOpen })}
                />
                <Menu
                  right
                  width="100%"
                  isOpen={isOpen}
                  className={classes.burgerMenu}
                  customBurgerIcon={false}
                  customCrossIcon={false}
                  noOverlay
                  data-test-id="slideOutMenu"
                >
                  <FiX
                    className={classes.menuIcon}
                    data-test-id="closeMenuButton"
                    onClick={() => this.setState({ isOpen: false })}
                  />
                  <nav className={classes.responsiveMenuItems}>
                    <MenuItems />
                  </nav>
                  <div className={classes.menuFooterWrapper}>
                    <img src={menuLogo} alt="Kued Logo" className={classes.menuLogo} />
                    <Text bold style={{ color: 'white' }}>Kued for private servers.</Text>
                    <Text caption bold style={{ color: 'white' }}>v1.0.0</Text>
                  </div>
                </Menu>
              </Fragment>
            ) : (
              <nav className={classes.menuItems}>
                <MenuItems />
              </nav>
            )
        }
      </Fragment>
    );
  }
}

export default TopNavigation;
