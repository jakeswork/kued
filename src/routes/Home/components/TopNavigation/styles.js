export default {
  menuItems: {
    position: 'fixed',
    top: 0,
    right: 0,
    display: 'flex',
    left: '50%',
    alignItems: 'flex-end',
    justifyContent: 'space-around',
  },
  menuFooterWrapper: {
    position: 'absolute',
    bottom: 32,
    right: 0,
    left: 0,
    flexDirection: 'column',
    display: 'flex !important',
    alignItems: 'center',
  },
  menuLogo: {
    width: 80,
  },
  responsiveMenuItems: {
    display: 'flex !important',
    flexDirection: 'column',
    padding: '64px 16px',
  },
  burgerMenu: {
    top: 0,
    background: 'rgba(0,0,0,0.9)',
  },
  menuIcon: {
    fontSize: 24,
    position: 'fixed',
    top: 16,
    right: 16,
    cursor: 'pointer',
    color: 'white',
  },
  img: {
    position: 'fixed',
    left: 16,
    top: '0.84375rem',
    width: 160,
  },
};
