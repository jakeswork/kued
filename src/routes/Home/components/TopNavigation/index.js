import withStyles from 'react-jss';

import TopNavigation from './TopNavigation';
import styles from './styles';

export default withStyles(styles)(TopNavigation);
