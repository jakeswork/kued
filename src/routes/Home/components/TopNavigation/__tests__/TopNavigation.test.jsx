import React from 'react';
import { shallow } from 'enzyme';

import TopNavigation from '../TopNavigation';

const defaultProps = {
  classes: {},
};

describe('The TopNavigation instance', () => {
  describe('When called with default properties', () => {
    it('should render without throwing errors', () => {
      const wrapper = shallow(<TopNavigation {...defaultProps} />);

      expect(wrapper).toMatchSnapshot();
    });
  });
});
