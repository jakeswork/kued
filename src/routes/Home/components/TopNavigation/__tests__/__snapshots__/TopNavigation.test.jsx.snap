// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The TopNavigation instance When called with default properties should render without throwing errors 1`] = `
<Fragment>
  <img
    alt="Kued Logotype"
    src="kued-logotype.png"
  />
  <nav>
    <Jss(MenuItems) />
  </nav>
</Fragment>
`;
